src/runtime.js

```diff
@@ -151,6 +151,7 @@ let tornado = {
    */
   exists(val, placeholderNode, bodies, context) {
     if (this.util.isPromise(val)) {
+      placeholderNode = this.insertPendingBody(placeholderNode, bodies.pending, context) || placeholderNode;
```

Comment:
Okay, this all may be premature optimization. I don't think there are any other insertion points in the lifecycle, so adding more "standardized" bodies probably won't be a problem.

Having said that, I suppose I can imagine a `reject` or `fail` body to replace `else` when a promise rejects. But, that would be in a completely different part of the lifecycle.

Do you think this needs to be resolved before we pull this PR?